helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataReader.java

```diff
@@ -58,25 +58,9 @@ public ZkRoutingDataReader(String namespace, String zkAddress,
     _zkClient = DedicatedZkClientFactory.getInstance()
         .buildZkClient(new HelixZkClient.ZkConnectionConfig(zkAddress),
             new HelixZkClient.ZkClientConfig().setZkSerializer(new ZNRecordSerializer()));
-
-    // Ensure that ROUTING_DATA_PATH exists in ZK. If not, create
-    // create() semantic will fail if it already exists
-    try {
-      _zkClient.createPersistent(MetadataStoreRoutingConstants.ROUTING_DATA_PATH, true);
-    } catch (ZkNodeExistsException e) {
-      // This is okay
-    }
```

Comment:
Why is this section removed? In a scenario where `ZkRoutingDataReader` is instantiated by itself, `subscribeRoutingDataChanges` might encounter `NoNodeException` if we don't ensure that ROUTING_DATA_PATH exists in ZK.

Reply:
Adding it back.